src/Microsoft.SqlTools.ServiceLayer/Scripting/ScriptAsScriptingOperation.cs

```diff
@@ -0,0 +1,322 @@
+                scriptingOptions.TargetServerVersion = SqlServerVersion.Version130;
+            }
+            else if (this.Parameters.ScriptOptions.ScriptCompatibilityOption == SqlScriptOptions.ScriptCompatabilityOptions.Script120Compat.ToString())
+            {
+                scriptingOptions.TargetServerVersion = SqlServerVersion.Version120;
+            }
+            else if (this.Parameters.ScriptOptions.ScriptCompatibilityOption == SqlScriptOptions.ScriptCompatabilityOptions.Script110Compat.ToString())
+            {
+                scriptingOptions.TargetServerVersion = SqlServerVersion.Version110;
+            }
+            else if (this.Parameters.ScriptOptions.ScriptCompatibilityOption == SqlScriptOptions.ScriptCompatabilityOptions.Script105Compat.ToString())
+            {
+                scriptingOptions.TargetServerVersion = SqlServerVersion.Version105;
+            }
+            else if (this.Parameters.ScriptOptions.ScriptCompatibilityOption == SqlScriptOptions.ScriptCompatabilityOptions.Script100Compat.ToString())
+            {
+                scriptingOptions.TargetServerVersion = SqlServerVersion.Version100;
+            }
+            else if (this.Parameters.ScriptOptions.ScriptCompatibilityOption == SqlScriptOptions.ScriptCompatabilityOptions.Script90Compat.ToString())
+            {
+                scriptingOptions.TargetServerVersion = SqlServerVersion.Version90;
+            }
+            else
+            {
+                //If you are getting this assertion fail it means you are working for higher
+                //version of SQL Server. You need to update this part of code.
+                 Logger.Write(LogLevel.Warning, "This part of the code is not updated corresponding to latest version change");
+            }
+
+            // for cloud scripting to work we also have to have Script Compat set to 105.
+            // the defaults from scripting options should take care of it
+            object targetDatabaseEngineType;
+            if (Enum.TryParse(typeof(SqlScriptOptions.ScriptDatabaseEngineType), this.Parameters.ScriptOptions.TargetDatabaseEngineType, out targetDatabaseEngineType))
+            {
+                switch ((SqlScriptOptions.ScriptDatabaseEngineType)targetDatabaseEngineType)
+                {
+                    case SqlScriptOptions.ScriptDatabaseEngineType.SingleInstance:
+                        scriptingOptions.TargetDatabaseEngineType = DatabaseEngineType.Standalone;
+                        break;
+                    case SqlScriptOptions.ScriptDatabaseEngineType.SqlAzure:
+                        scriptingOptions.TargetDatabaseEngineType = DatabaseEngineType.SqlAzureDatabase;
+                        break;
+                }
+            }
+
+            object targetDatabaseEngineEdition;
+            if (Enum.TryParse(typeof(SqlScriptOptions.ScriptDatabaseEngineEdition), this.Parameters.ScriptOptions.TargetDatabaseEngineEdition, out targetDatabaseEngineEdition))
```

Comment:
Are we missing settings in Ops Studio for controlling this? We should create tracking tasks for all this kind of work. There are a whole bunch of script options and users are starting to request them one by one. We should probably do the work in one batch